### DISCUSSION TITLE
build version auto increment with IONIC

### DISCUSSION THREAD

**Discussion by schhatbar** [codemagic-ci-cd/codemagic-docs#1022] [Q&A] [answered]
```
script: |
        LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number APP_STORE_APP_ID)
        cd $APP_PATH/ios/App # avgtool must run in the folder where xcodeproj file is located
        agvtool new-version -all $(($LATEST_BUILD_NUMBER + `1))```
```

On above script `app-store-connect get-latest-app-store-build-number APP_STORE_APP_ID` statement isn't returning any version , so end result is always goes to "1" and I don't see any way to debug this statement in Codemegic , 

It would be great help if someone have any Idea about how it works for Ionic incremental build version.

Let me know if you required more details

**Comment by himesh-cm** [comment, marked as answer]
@schhatbar Can you confirm if you are using $APP_STORE_APP_ID?

**Reply by schhatbar** [reply]
@himesh-cm I have crosschecked it multiple times.

**Reply by himesh-cm** [reply]
Okay, in the script you shared there is no $ sign so I confirmed.
can you confirm if there is a version available on the apple play store?

**Reply by schhatbar** [reply]
I have defined it to as yaml environment variables

**Reply by schhatbar** [reply]
```
workflows:
  ionic-capacitor-ios-workflow:
        name: Ionic Capacitor iOS Workflow
        max_build_duration: 120
        instance_type: mac_mini
        environment:
            groups:
            # Add the group environment variables in Codemagic UI (either in Application/Team variables) - https://docs.codemagic.io/variables/environment-variable-groups/
              # - manual_code_signing # <-- (Includes CM_CERTIFICATE, CM_CERTIFICATE_PASSWORD, CM_PROVISIONING_PROFILE)
                - ai_team 
            vars:
                # Ionic Xcode worskspace and scheme
                XCODE_WORKSPACE: "ios/App/App.xcworkspace"
                XCODE_SCHEME: "App"
                APP_STORE_APP_ID: 6443677258 # <-- Put the app id number here. This is found in App Store Connect > App > General > App Information
            node: v16.11.1
            xcode: 13.3
            cocoapods: default
            ios_signing:
              provisioning_profiles:
                - profile: aiteamcert
              certificates:
                - certificate: Certificate
        triggering:
            events:
                - push
            branch_patterns:
                - pattern: develop
                  include: true
                  source: true
        scripts:
            - name: Goto App Path
              script: |
                cd $APP_PATH
            - name: Install npm dependencies for Ionic project
              script: |
                cd $APP_PATH && npm install && ionic build --prod && npx cap add ios && npx cap sync ios
            - name: Cocoapods installation
              script: |
                cd $APP_PATH/ios/App && pod install
            - name: Update dependencies and copy web assets to native project
              script: |
                # npx cap copy # <- use this is you don't need to update native dependencies
                cd $APP_PATH && npx cap sync # <- update native dependencies and copy web assets to native project
            - name: Set up keychain to be used for codesigning using Codemagic CLI 'keychain' command
              script: |
                keychain initialize
            # - name: Set up Provisioning profiles from environment variables (Use with manual code signing)
            #   script: |
            #     PROFILES_HOME="$HOME/Library/MobileDevice/Provisioning Profiles"
            #     mkdir -p "$PROFILES_HOME"
            #     PROFILE_PATH="$(mktemp "$PROFILES_HOME"/$(uuidgen).mobileprovision)"
            #     echo ${CM_PROVISIONING_PROFILE} | base64 --decode > "$PROFILE_PATH"
            #     echo "Saved provisioning profile $PROFILE_PATH"
            #- name: Fetch signing files
            #  script: | 
            #    app-store-connect fetch-signing-files "$BUNDLE_ID" \ --type IOS_APP_STORE \ --create --private-key $APP_STORE_CONNECT_PRIVATE_KEY
            #    app-store-connect fetch-signing-files $(xcode-project detect-bundle-id) --type IOS_APP_STORE --create --private-key $APP_STORE_CONNECT_PRIVATE_KEY
            # - name: Set up signing certificate (Use with manual code signing)
            #   script: |
            #     echo $CM_CERTIFICATE | base64 --decode > /tmp/certificate.p12
            #     keychain add-certificates --certificate /tmp/certificate.p12 --certificate-password $CM_CERTIFICATE_PASSWORD
           
            - name: Add certificates to keychain
              script: |
                keychain add-certificates
            - name: Increment build number $BUILD_NUMBER
              script: |
                LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number $APP_STORE_APP_ID)
                cd $APP_PATH/ios/App # avgtool must run in the folder where xcodeproj file is located
                agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
            - name: Set up code signing settings on Xcode project
              script: |
                xcode-project use-profiles
            - name: Build ipa for distribution
              script: |
                xcode-project build-ipa --workspace "$APP_PATH/$XCODE_WORKSPACE" --scheme "$XCODE_SCHEME"
        artifacts:
            - build/ios/ipa/*.ipa
            - /tmp/xcodebuild_logs/*.log
            - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.app
            - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
        publishing:
          app_store_connect:
              api_key: $APP_STORE_CONNECT_PRIVATE_KEY         # Contents of the API key, can also reference environment variable such as $APP_STORE_CONNECT_PRIVATE_KEY
              key_id: $APP_STORE_CONNECT_KEY_IDENTIFIER       # Alphanumeric value that identifies the API key, can also reference environment variable such as $APP_STORE_CONNECT_KEY_IDENTIFIER
              issuer_id: $APP_STORE_CONNECT_ISSUER_ID         # Alphanumeric value that identifies who created the API key, can also reference environment variable such as $APP_STORE_CONNECT_ISSUER_ID
              submit_to_testflight: false                     # Optional boolean, defaults to false. Whether or not to submit the uploaded build to TestFlight beta review. Required for distributing to beta groups. Note: This action is performed during post-processing.
              # beta_groups:                                  # Specify the names of beta tester groups that will get access to the build once it has passed beta review. 
              #   - group name 1
              #   - group name 2
          email:
            recipients:
                - 
            notify:
                success: false     # To not receive a notification when a build succeeds
                failure: false     # To not receive a notification when a build fails
```

Here is whole yaml file

All other things are working perfectly  rather then getting testflight version from app store.

**Reply by himesh-cm** [reply]
what is the version of the app which is available on testflight?

**Reply by schhatbar** [reply]
It's 1.0 and build version is 1
It worked for first testflight but after that it isn't returning any version from it.

**Reply by himesh-cm** [reply]
can you try the below script once
```
#!/bin/sh
set -e
set -x
cd $FCI_BUILD_DIR/ios/App
agvtool new-version -all $(($(app-store-connect get-latest-testflight-build-number "$APP_STORE_APP_ID") + 1))
```

**Reply by schhatbar** [reply]
I have tried it too . It's not working either . Still let me try again.

**Reply by schhatbar** [reply]
@himesh-cm  Thanks for quick response !! It worked.

**Reply by himesh-cm** [reply]
@schhatbar, please mark it as an answer, so others may find it helpful too.
